src/Symfony/Component/HttpKernel/EventListener/ExceptionListener.php

```diff
@@ -37,22 +37,30 @@ class ExceptionListener implements EventSubscriberInterface
     protected $debug;
     private $charset;
     private $fileLinkFormat;
+    private $silentIgnore;
 
-    public function __construct($controller, LoggerInterface $logger = null, $debug = false, $charset = null, $fileLinkFormat = null)
+    public function __construct($controller, LoggerInterface $logger = null, $debug = false, $charset = null, $fileLinkFormat = null, array $silentIgnore = array())
     {
         $this->controller = $controller;
         $this->logger = $logger;
         $this->debug = $debug;
         $this->charset = $charset;
         $this->fileLinkFormat = $fileLinkFormat;
+        $this->silentIgnore = $silentIgnore;
     }
 
     public function logKernelException(GetResponseForExceptionEvent $event)
     {
         $exception = $event->getException();
-        $request = $event->getRequest();
+        do {
+            foreach ($this->silentIgnore as $class) {
+                if (is_a($exception, $class)) {
+                    return;
+                }
+            }
+        } while (null !== $exception = $exception->getPrevious());
```

Comment:
use a different variable name for this loop, to avoid overriding the one containing the main exception